pkg/dataloader/jiraloader/jiraloader.go

```diff
@@ -0,0 +1,354 @@
+	var resolutionTime *time.Time
+
+	changelogLayout := "2006-01-02T15:04:05.999-0700"
+
+	for _, history := range issue.Changelog.Histories {
+		for _, item := range history.Items {
+			// If trt-incident label was removed from a jira ticket, consider that the time when the incident
+			// was resolved.
+			if !issueContainsLabel(issue, "trt-incident") && item.Field == "labels" && item.FromString == "trt-incident" && item.ToString != "trt-incident" {
+				createdTime, err := time.Parse(changelogLayout, history.Created)
+				if err != nil {
+					log.WithError(err).Warningf("parsing error: %s", history.Created)
+					continue
+				}
+				// We pick the oldest time we removed the trt-incident label (maybe we toggled back and forth a few
+				// times).
+				if resolutionTime == nil || resolutionTime.Before(createdTime) {
+					log.Debugf("trt-incident label was removed from %s at %+v", issue.Key, createdTime)
+					resolutionTime = &createdTime
+				}
+			}
+
+			// OCPBUGS don't get a resolution time until it's closed which happens when a release GA's. We
+			// find the first terminal incident status changelog instead. From TRT's perspective, we don't
+			// care about OCPBUGS incidents after they go to MODIFIED.
+			resolvedStatuses := []string{"MODIFIED", "ON_QA", "Verified", "Closed"}
+			for _, status := range resolvedStatuses {
+				if item.ToString == status {
+					createdTime, err := time.Parse(changelogLayout, history.Created)
+					if err != nil {
+						log.WithError(err).Warningf("parsing error: %s", history.Created)
+						continue
+					}
+					// We pick the oldest state change
+					if resolutionTime == nil || resolutionTime.After(createdTime) {
+						log.Debugf("%s to %s at %+v", issue.Key, status, createdTime)
+						resolutionTime = &createdTime
+					}
+				}
+			}
+		}
+	}
+
+	// Fallback to the jira resolution time
+	if issue.Fields.ResolutionDate != "" && resolutionTime == nil {
+		jiraResolutionTime, err := time.Parse(jiraTimeLayout, issue.Fields.ResolutionDate)
+		if err != nil {
+			fmt.Printf("parsing error: %+v", err)
+		}
+		log.Debugf("resolution time for %s is %+v", issue.Key, jiraResolutionTime)
+		resolutionTime = &jiraResolutionTime
+	}
+
+	return resolutionTime
+}
+
+func issueContainsLabel(issue *v1jira.Issue, label string) bool {
+	for _, issueLabel := range issue.Fields.Labels {
+		if issueLabel == label {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (jl *JiraLoader) componentLoader() {
+	start := time.Now()
+	log.Infof("loading jira ocpbugs component information...")
+	body, err := jiraRequest("https://issues.redhat.com/rest/api/2/project/12332330/components")
+	if err != nil {
+		jl.errors = append(jl.errors, err)
+		return
+	}
+
+	var components []v1jira.Component
+	err = json.Unmarshal(body, &components)
+	if err != nil {
+		jl.errors = append(jl.errors, err)
+		return
+	}
+
+	tblUpdater := stagingtablemanager.New(jl.dbc, "jira_components")
+	tmpTable, err := tblUpdater.CreateStagingTable()
```

Comment:
Should you do a Cleanup prior to each Create? (in case process got killed or something)

Reply:
The tables get a timestamp appended,  so another downside to this approach is potentially need to clean-up that if it happens